esmvalcore/preprocessor/_io.py

```diff
@@ -171,12 +171,22 @@ def _get_concatenation_error(cubes):
     raise ValueError(f'Can not concatenate cubes: {msg}')
 
 
+def _fix_cube_endianess(cubes):
+    """Transform cubes in big endian to little"""
+    for cube in cubes:
+        if cube.dtype.byteorder == ">":
+            logger.warning("Changing cube endianess to little. This may be "
+                           "memory intensive.")
+            cube.data = cube.data.byteswap().newbyteorder()
```

Comment:
this is a flat-out data realization in memory, could you try exploring the use of `core_data()` instead of `data` please? That way we keep things lazy